my-app/src/App.js

```diff
@@ -5,51 +5,51 @@ import Dashboard from './components/Dashboard';
 
 class App extends React.Component {
   state = {
-    ball: 0,
-    strike: 0
+    balls: 0,
+    strikes: 0
   }
 
   handleStrike = () => {
-
-    if (this.state.strike < 2) {
-      this.setState(prevState => {
-        return {strike: prevState.strike + 1}
-     })
-    } else {
-      this.setState({strike: 0, ball: 0})
-    }
+    this.setState(prevState => {      
+      // if less than 2 strikes, increment
+      if (prevState.strikes < 2) {
+        return {strikes: prevState.strikes + 1}
+      } else {
+        return {strikes: 0, balls: 0}
+      }
+     })    
   }
 
   handleBall = () => {
-    if (this.state.ball < 3) {
-      this.setState(prevState => {
-        return {ball: prevState.ball + 1}
+    this.setState(prevState => {      
+      if (prevState.balls < 3) {
+        return {balls: prevState.balls + 1}
+      } else {
+        return {strikes: 0, balls: 0}
+      }
       })
-    } else {
-      this.setState({strike: 0, ball: 0})
-    }
   }
 
   handleHit = () => {
-    this.setState({strike: 0, ball: 0})
+    this.setState({strikes: 0, balls: 0})
   }
 
   handleFoul = () => {
 
     this.setState(prevState => {
 
```

Comment:
foul is one strike, it increases up to 2 strikes, i.e with one strike a foul would make it 2 strikes